src/Frontend/reducers/locale/localeReducer.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * This file is part of the Sententiaregum project.
+ *
+ * (c) Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ * (c) Ben Bieler <benjaminbieler2014@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import { GET_LOCALES, CHANGE_LOCALE }   from '../../constants/Locale';
+import Locale                           from '../../util/http/Locale';
+import ApiKey                           from '../../util/http/ApiKey';
+import axios                            from 'axios';
+
+const localeReducer = (state = [], action) => {
+  switch (action.type) {
+  case GET_LOCALES:
+    return state;
+
+  case CHANGE_LOCALE:
+    const locale = action.locale;
+
+    Locale.setLocale(locale);
+      //TODO: replace this useless statement.
+    if (true) {
+      //TODO: Fix auth
+      axios.patch('/api/protected/locale.json', { locale }, {
```

Comment:
Please move such logic (`axios.*` or `Locale.setLocale`) to an action creator (http://redux.js.org/docs/advanced/AsyncActions.html). Reducer function should be side-effect free (purity of functions) and calls against a REST API are definetely a side-effect.